### PR TITLE
Bug 1813994 - Cleanup deprecated mobile products

### PR DESCRIPTION
**Pull request by gbrownmozilla**
A more conservative replacement for #1295

### CODE REVIEW

File: api/src/shipit_api/admin/release.py [hunk 1]
```diff
@@ -37,7 +37,11 @@ def parse_version(product, version):
         except KeyError:
             raise ValueError(f"Product {product} versions are not supported")
 
-    VersionClass = _VERSION_CLASS_PER_PRODUCT[product_enum]
+    try:
```
Comment (gbrownmozilla) [inline api/src/shipit_api/admin/release.py:40]: Left this in, copied from #1295: even though nothing else is changing here, this seems like a good idea.

File: api/src/shipit_api/admin/settings.py [hunk 2]
```diff
@@ -75,11 +75,8 @@
 LDAP_GROUPS = {
     "admin": ADMIN_LDAP_GROUP,
     "firefox-signoff": ["shipit_firefox"],
-    "fenix-signoff": ["shipit_mobile"],
```
Comment (gbrownmozilla) [inline api/src/shipit_api/admin/settings.py:78]: Should be no need to signoff on old products.

File: api/src/shipit_api/public/api.yml
Comment (JohanLorenzo) [inline api/src/shipit_api/public/api.yml:1]: Question: I see we still allow deprecated products in the `Release` scheme, unlike what we did in the original PR[1]. What kind of issues this part of the original PR (would have) caused? 

[1] https://github.com/mozilla-releng/shipit/pull/1295/files#diff-872926f584b15572d240a25ce24bf884b1cd0cb584bac73020b7c4cf4a47e390

**Review by JohanLorenzo — APPROVED** [review]
Looks good to me! I agree with this more conservative approach. I look forward to seeing this cleanup in action 😃 

I mentally diff'd this current PR with the original one (https://github.com/mozilla-releng/shipit/pull/1295). If I understand correctly and for the record, they're the same expect these files didn't get changed:

 * api/src/shipit_api/admin/release.py 
 * api/src/shipit_api/public/api.py
 * api/tests/test_release.py

**Review by gabrielBusta — APPROVED** [review]

**Pull request merged**